tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+	}
+
+	public function testFulltextExtractionText() {
+		$this->_addFileToDocument( 'test.txt', 'TXT fulltext' );
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testFulltextExtractionWithNonExistentFile() {
+		$doc = new Opus_Document( $this->document_id );
+
+		$file = $doc->addFile();
+		$file->setPathName( 'nonexistent.pdf' );
+		$file->setLabel( 'non-existent PDF fulltext' );
+
+		$doc->store();
+
+		$this->indexer->addDocumentsToIndex( $doc );
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testFulltextExtractionWithNonSupportedMimeType() {
+		$this->_addFileToDocument( 'test.odt', 'ODT fulltext' );
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testFulltextExtractionByContentForPdf() {
+		$this->_addFileToDocument( 'test.pdf', 'PDF fulltext' );
+
+		$this->assertEquals( 1, $this->_searchTestFulltext() );
+	}
+
+	public function testFulltextExtractionByContentForPostscript() {
+		$this->markTestIncomplete();
+
+		$this->_addFileToDocument( 'test.ps', 'PS fulltext' );
+
+		$this->assertEquals( 1, $this->_searchTestFulltext() );
+	}
+
+	public function testFulltextExtractionByContentForText() {
+		$this->_addFileToDocument( 'test.txt', 'TXT fulltext' );
+
+		$this->assertEquals( 1, $this->_searchTestFulltext() );
+	}
+
+	public function testFulltextExtractionByContentForHtml() {
+		$this->_addFileToDocument( 'test.html', 'HTML fulltext' );
+
+		$this->assertEquals( 1, $this->_searchTestFulltext() );
+	}
+
+	public function testFulltextExtractionByContentForXhtml() {
+		$this->_addFileToDocument( 'test.xhtml', 'XHTML fulltext' );
+
+		$this->assertEquals( 1, $this->_searchTestFulltext() );
+	}
+
+	private function _getNumberOfIndexDocs() {
+		$search = Opus_Search_Service::selectSearchingService();
+
+		return $search->customSearch( $search->createQuery() )
+		              ->getAllMatchesCount();
+	}
+
+	private function _searchTestFulltext() {
+		$search = Opus_Search_Service::selectSearchingService();
+		$query  = $search->createQuery()->setFilter(
+			$search->createFilter()
+			       ->addFilter( Opus_Search_Filter_Simple::createCatchAll( 'Lorem' ) )
+		);
+
+		return $search->customSearch( $query )->getAllMatchesCount();
+	}
+
+	private function _addOneDocumentToIndex() {
+		$document = new Opus_Document( $this->document_id );
+		$this->indexer->addDocumentsToIndex( $document );
+	}
+
+	/**
+	 *
+	 * @param string $filename
+	 * @param string $label
+	 */
+	private function _addFileToDocument( $filename, $label ) {
+		$doc  = new Opus_Document( $this->document_id );
+		$file = $doc->addFile();
+		$file->setTempFile( APPLICATION_PATH . '/tests/fulltexts/' . $filename );
+		$file->setPathName( $filename );
+		$file->setLabel( $label );
+		$file->setVisibleInFrontdoor( '1' );
+
+		$doc->store();
+
+		$this->indexer->addDocumentsToIndex( $doc );
+	}
+
+	public function testAttachFulltextToNull() {
```

Comment:
Communicating using text can be difficult and I think sometimes we make it even harder by using English. The function tested accepted 'null' as second argument. Apparently the function itself did not throw an exception although looking at it I think array_filter should have thrown something. I cannot tell you what the original motivation of the developer was who wrote the test. I think it simply that some documents do not have files and some code might have been written in a way that does not check this first before trying to attach "null" files to the XML for the indexer. I do not know what else to say.